opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -674,4 +995,163 @@ mod tests {
         assert!(!log_batch.data.is_empty()); // Should have encoded data
         assert_eq!(log_batch.metadata.schema_ids.matches(';').count(), 0); // 1 schema = 0 semicolons
     }
+
+    #[test]
+    fn test_span_encoding() {
+        let encoder = OtlpEncoder::new();
+
+        let mut span = Span {
+            trace_id: vec![1; 16],
+            span_id: vec![2; 8],
+            parent_span_id: vec![3; 8],
+            name: "test_span".to_string(),
+            kind: 1, // CLIENT
+            start_time_unix_nano: 1_700_000_000_000_000_000,
+            end_time_unix_nano: 1_700_000_001_000_000_000,
+            flags: 1,
+            trace_state: "key=value".to_string(),
+            ..Default::default()
+        };
+
+        // Add some attributes
+        span.attributes.push(KeyValue {
+            key: "http.method".to_string(),
+            value: Some(AnyValue {
+                value: Some(Value::StringValue("GET".to_string())),
+            }),
+        });
+
+        span.attributes.push(KeyValue {
+            key: "http.status_code".to_string(),
+            value: Some(AnyValue {
+                value: Some(Value::IntValue(200)),
+            }),
+        });
+
+        let metadata = "namespace=testNamespace/eventVersion=Ver1v0";
+        let result = encoder.encode_span_batch([span].iter(), metadata).unwrap();
+
+        assert!(!result.is_empty());
```

Comment:
Use a stricter assertion.

```suggestion
        assert_eq!(result.len(), 1);
```